### PR TITLE
Transaction handling on expense dashboard

### PR DESCRIPTION
**Pull request by szattila98**

### CODE REVIEW

**Comment by djavorek** [comment]
Code looks good to me at this point. Will check it again after it can be tested.

**Comment by szattila98** [comment]
Implemented functionality. I could not find a date picker component so there is only a TextField for it yet. 
When most of the functionalities are implemented we should create issues for planning and implementing validations and error handling where it is non-existant or not according to planning.

File: client/src/lib/components/NewTransactionDialog.svelte [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+	const date = field('date', '', [required()]);
+	const newTransactionForm = form(donorName, value, date);
+
+	function onSubmit() {
+		if (
+			$newTransactionForm.valid &&
+			$donorName.value &&
+			selectedCurrencyTypeId &&
+			$value.value &&
+			$date.value
```
Comment (djavorek) [inline client/src/lib/components/NewTransactionDialog.svelte:30]: Is it possible for the form to be valid but these values are missing?
I would have probably only check the first one.

File: client/src/routes/dashboard.svelte [hunk 2]
```diff
@@ -7,29 +7,88 @@
...
+	onMount(() => {
+		fetch(`/api/expense/${userId}`, {
+			method: 'GET',
+			headers: {
+				'Content-Type': 'application/json',
+			},
+		}).then((res) => {
+			res.json().then((res) => {
+				expenses = res.data.expenses as Expense[];
+				clickedExpense = expenses && expenses[0] ? expenses[0] : null;
```
Comment (djavorek) [inline client/src/routes/dashboard.svelte:48]: Do we want to select the first expense by default (or leave it blank until the user selects one)?
If not, then this is not necessary anymore.
$: clickedExpense = .. line does the trick

**Review by djavorek — APPROVED** [review]
Two minor comments. Can be merged without those.
Good job, lot of work 💥

**Comment by djavorek** [comment]
> Implemented functionality. I could not find a date picker component so there is only a TextField instead of that. When most of the functionalities are implemented we should create issues for planning and implementing validations and error handling where it is non-existant or not according to planning.

Yep, found this issue for the datepicker thing: https://github.com/hperrin/svelte-material-ui/issues/268
It looks like they even disabled the default html5 datepicker, to be able to implement their own version, but the work has not even started yet.
It does not matter to us that much.

File: client/src/lib/components/NewTransactionDialog.svelte (hunk 1 above)
Comment (szattila98) [reply client/src/lib/components/NewTransactionDialog.svelte]: For sure, I was a little blind for the inner workings of the validator library, it is much more streamlined now.

File: client/src/routes/dashboard.svelte (hunk 2 above)
Comment (szattila98) [reply client/src/routes/dashboard.svelte]: I figured, the first should always be showing, it's only a minor tweak to make if necessary, let's leave it like this for now.
Nonetheless I streamlined the promise-then forest that was present there with the much sleeker async design.

File: client/src/lib/components/NewTransactionDialog.svelte (hunk 1 above)
Comment (szattila98) [reply client/src/lib/components/NewTransactionDialog.svelte]: Yes I was a little blind about the inner workings of the validator library, it is more clean now.

File: client/src/lib/components/NewTransactionDialog.svelte [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+	const donorName = field('donorName', '', [required()]);
+	const selectedCurrencyTypeId = field('selectedCurrencyTypeId', 0, [required()]);
+	const value = field('value', 0, [required()]);
+	const date = field('date', '', [required()]);
+	const newTransactionForm = form(donorName, value, date);
+
+	async function onSubmit() {
+		await newTransactionForm.validate();
+		if ($newTransactionForm.valid) {
+			console.log('nein');
```
Comment (djavorek) [inline client/src/lib/components/NewTransactionDialog.svelte:27]: Excuse me? 😂

**Review by djavorek — APPROVED** [review]

**Pull request merged**